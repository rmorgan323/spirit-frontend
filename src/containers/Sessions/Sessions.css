/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.Sessions .sessions-main-holder {
  width: 675px;
  margin: 20px auto; }

.Sessions .session {
  display: flex;
  align-items: center;
  border-bottom: 10px solid #f1f2f2; }

.Sessions .session-number {
  width: 120px;
  font-size: 20px; }

.Sessions .session-date {
  width: 290px;
  margin-right: 20px;
  font-size: 20px; }

.Sessions .date-span {
  color: #a7a9ac;
  font-family: "Josefin Sans", sans-serif;
  font-size: 14px;
  font-weight: 900;
  font-size: 16px; }

.Sessions .session-status {
  margin-right: 20px;
  width: 120px;
  font-size: 20px; }

.Sessions .select-session-button {
  position: relative;
  top: 4px;
  width: 100px;
  border: none;
  background: none;
  color: #5cba47;
  text-decoration: none;
  cursor: pointer; }

.Sessions .select-session-button:hover {
  color: #468c41; }

.Sessions .join-link button {
  width: 100px;
  height: 40px;
  padding: 0 10px;
  border: none;
  background-color: #5cba47;
  color: #fff;
  cursor: pointer; }

.Sessions .join-link button:hover {
  background-color: #468c41; }

.Sessions .create-session-button {
  width: 300px;
  height: 40px;
  margin: 30px auto;
  display: block;
  background-color: #9E87BA;
  color: #fff;
  cursor: pointer;
  border: none;
  transition: all .2s ease-in-out; }

.Sessions .create-session-button:hover {
  background-color: #5E378C; }

.Sessions .compare-button-holder {
  display: block;
  position: relative;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  user-select: none;
  margin-right: 2px;
  width: 30px;
  top: 2px; }

.Sessions .compare-button-holder input {
  position: absolute;
  opacity: 0;
  cursor: pointer; }

.Sessions .checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 13px;
  width: 13px;
  background-color: #f1f2f2;
  border-radius: 50px;
  border: 2px solid #d1d3d4;
  box-shadow: inset 0px 0px 4px -1px rgba(0, 0, 0, 0.77);
  transition: all .2s ease-in-out; }

.Sessions .compare-button-holder:hover input ~ .checkmark {
  background-color: #d1d3d4; }

.Sessions .compare-button-holder input:checked ~ .checkmark {
  background-color: #a8ed8a; }

.Sessions .checkmark:after {
  content: "";
  position: absolute;
  display: none; }

.Sessions .compare-button-holder input:checked ~ .checkmark:after {
  display: block; }

.Sessions .button-holder {
  height: 40px;
  width: 100%;
  position: relative;
  margin-bottom: 26px; }

.Sessions .compare-sessions-button {
  position: absolute;
  bottom: 0;
  left: 0;
  height: 40px;
  background-color: #5cba47;
  font-size: 14px;
  width: 88px;
  border: none;
  color: #fff;
  transition: all .2s ease-in-out;
  cursor: pointer; }

.Sessions .compare-sessions-button:hover {
  background-color: #2e5b2a; }

.Sessions .goals-chart {
  width: 675px;
  margin: 20px auto; }
