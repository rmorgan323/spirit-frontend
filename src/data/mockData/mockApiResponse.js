const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;