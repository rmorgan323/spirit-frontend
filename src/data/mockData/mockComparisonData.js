/*eslint-disable camelcase*/
/*eslint-disable id-length*/

export const mockComparisonData = [
  {
    created: [
      {
        date: '2/26/2018'
      },
      {
        date: '2/24/2018'
      }
    ]
  },
  {
    session_id: [
      {
        id: 54
      },
      {
        id: 53
      }
    ]
  },
  {
    executive: [
      {
        x: 1,
        y: 22
      },
      {
        x: 0,
        y: 20
      }
    ]
  },
  {
    modulation: [
      {
        x: 1,
        y: 18
      },
      {
        x: 0,
        y: 12
      }
    ]
  },
  {
    postural: [
      {
        x: 1,
        y: 15
      },
      {
        x: 0,
        y: 8
      }
    ]
  },
  {
    sensory: [
      {
        x: 1,
        y: 28
      },
      {
        x: 0,
        y: 22
      }
    ]
  },
  {
    social: [
      {
        x: 1,
        y: 18
      },
      {
        x: 0,
        y: 13
      }
    ]
  }
];

