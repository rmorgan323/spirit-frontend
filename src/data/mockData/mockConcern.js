/*eslint-disable camelcase*/

export const mockConcern = {
  id: 5,
  description: "Concern 1",
  domain_1: true,
  domain_2: true,
  domain_3: false,
  domain_4: true,
  domain_5: false,
  domain_6: false,
  notes: "Some notes.",
  patient_id: 5,
  created_at: "2018-02-20T20:08:19.142Z",
  updated_at: "2018-02-20T20:08:19.142Z"
};
