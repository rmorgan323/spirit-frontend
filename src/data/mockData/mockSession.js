/*eslint-disable camelcase*/

export const mockSession = {
  id: 10,
  concern_id: 5,
  created_at: "2018-02-20T21:25:45.411Z",
  updated_at: "2018-02-20T21:25:45.411Z",
  completed: true
};
