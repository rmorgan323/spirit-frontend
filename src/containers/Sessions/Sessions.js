/*eslint-disable id-length*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import moment from 'moment';
import GoalsChart from '../GoalsChart/GoalsChart';
import SessionsComparison from '../SessionsComparison/SessionsComparison';
import * as actions from '../../actions/index';
import './Sessions.css';

export class Sessions extends Component {
  constructor(props) {
    super(props);

    this.state = {
      checkedIds: []
    };
  }

  componentDidMount() {
    this.props.wipeStoreFromSessions();
  }

  toggleCheckBox = id => {
    const currentState = this.state.checkedIds;
    if (currentState.includes(id)) {
      let newState = currentState
        .filter(num => num !== id)
        .sort((a, b) => a - b);
      this.setState({ checkedIds: newState });
    } else {
      let newState = [...currentState, id].sort((a, b) => a - b);
      this.setState({ checkedIds: newState });
    }
  };

  checkedStatus = id => {
    const currentState = this.state.checkedIds;
    if (currentState.includes(id)) {
      return true;
    } else {
      return false;
    }
  };

  renderCheckBox = (id, completed) => {
    if (completed) {
      return (
        <label className="compare-button-holder">
          <input
            type="checkbox"
            checked={this.checkedStatus(id) ? true : false}
            onChange={() => this.toggleCheckBox(id)}
          />
          <span className="checkmark" />
        </label>
      );
    }
  };

  renderedSessions = () => {
    const renderThis = this.props.sessions.map(session => {
      return (
        <div className="session" key={session.id}>
          <div className="compare-button-holder">
            {this.renderCheckBox(session.id, session.completed)}
          </div>
          <p className="session-number">Session {session.id}</p>

          <p className="session-date">
            <span className="date-span">Last update: </span>
            {moment(session.updated_at).format('LL')}
          </p>

          <p className="session-status">
            {session.completed ? 'Completed' : 'In Progress'}
          </p>

          {!session.completed && (
            <Link to={`/spirit/sessions/${session.id}/sam`}>
              <button
                className="select-session-button"
                onClick={() => this.props.getSession(session)}
              >
                SELECT
              </button>
            </Link>
          )}

          {session.completed && (
            <Link
              className="join-link"
              to={`/spirit/sessions/${session.id}/view`}
            >
              <button
                className="session-result-button"
                onClick={() => this.props.getSession(session)}
              >
                RESULTS
              </button>
            </Link>
          )}
        </div>
      );
    });
    return renderThis;
  };

  compareButton = () => {
    if (this.state.checkedIds.length >= 2) {
      return (
        <button
          className="compare-sessions-button"
          onClick={() => this.props.compareSessionData(this.state.checkedIds)}
        >
          Compare Sessions
        </button>
      );
    }
  };

  compareTherapyGoals = () => {
    if (Object.keys(this.props.comparisonData).length) {
      return (
        <div>
          <div className="goals-chart">
            <GoalsChart />
          </div>
          <SessionsComparison />
        </div>
      );
    }
  };

  render() {
    return (
      <div className="Sessions">
        <div className="sessions-main-holder">
          <h2>Sessions</h2>
          {this.renderedSessions()}
          <div className="button-holder">
            <button
              onClick={() =>
                this.props.createSession(this.props.selectedConcern.id)
              }
              className="create-session-button"
            >
              Create New Session
            </button>
            {this.compareButton()}
          </div>
        </div>
        {this.compareTherapyGoals()}
      </div>
    );
  }
}

export const mapStateToProps = store => ({
  comparisonData: store.comparisonData,
  selectedConcern: store.selectedConcern,
  sessions: store.sessions
});

export const mapDispatchToProps = dispatch => ({
  createSession: selectedConcernId =>
    dispatch(actions.createSession(selectedConcernId)),
  getSession: selectedSessionId =>
    dispatch(actions.getSession(selectedSessionId)),
  compareSessionData: idArray => dispatch(actions.compareSessionData(idArray)),
  wipeStoreFromSessions: () => {
    dispatch(actions.wipeStoreFromSessions());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Sessions);

Sessions.propTypes = {
  selectedConcern: PropTypes.object,
  sessions: PropTypes.array,
  createSession: PropTypes.func,
  getSession: PropTypes.func,
  compareSessionData: PropTypes.func,
  comparisonData: PropTypes.array,
  wipeStoreFromSessions: PropTypes.func
};
