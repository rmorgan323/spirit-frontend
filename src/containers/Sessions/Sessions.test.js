import React from 'react';
import { Sessions, mapStateToProps, mapDispatchToProps } from './Sessions';
import { shallow } from 'enzyme';
import { mockConcern } from '../../data/mockData/mockConcern';
import { mockSession } from '../../data/mockData/mockSession';
import { mockComparisonData } from '../../data/mockData/mockComparisonData';

describe('Sessions tests', () => {
  let selectedConcern;
  let sessions;
  let createSession;
  let getSession;
  let compareSessionData;
  let comparisonData;
  let wipeStoreFromSessions;
  let renderedSessions;

  beforeEach(() => {
    selectedConcern = mockConcern;
    sessions = [mockSession];
    createSession = jest.fn();
    getSession = jest.fn();
    compareSessionData = jest.fn();
    comparisonData = mockComparisonData;
    wipeStoreFromSessions = jest.fn();

    renderedSessions = shallow(
      <Sessions
        selectedConcern={selectedConcern}
        sessions={sessions}
        createSession={createSession}
        getSession={getSession}
        compareSessionData={compareSessionData}
        comparisonData={comparisonData}
        wipeStoreFromSessions={wipeStoreFromSessions}
      />);
  });

  it('Should match the snapshot', () => {
    expect(renderedSessions).toMatchSnapshot();
  });

  it('should set state with an array when toggleCheckBox is called', () => {
    renderedSessions.instance().toggleCheckBox(1);
    
    expect(renderedSessions.state('checkedIds')).toEqual([1]);

    renderedSessions.instance().toggleCheckBox(1);

    expect(renderedSessions.state('checkedIds')).toEqual([]);

    renderedSessions.instance().toggleCheckBox(1);
    renderedSessions.instance().toggleCheckBox(2);

    expect(renderedSessions.state('checkedIds')).toEqual([1, 2]);
  });

  it('checkedStatus should return true or false depending on state array', () => {
    expect(renderedSessions.instance().checkedStatus(1)).toEqual(false);

    renderedSessions.instance().toggleCheckBox(1);

    expect(renderedSessions.instance().checkedStatus(1)).toEqual(true);
  });
});

describe('mapStateToPropsTests', () => {
  it('Should pull comparisonData, selectedConcern, and sessions from store', () => {
    const comparisonData = mockComparisonData;
    const selectedConcern = mockConcern;
    const sessions = [mockSession];
    const mockStore = { comparisonData, selectedConcern, sessions };
    const result = mapStateToProps(mockStore);

    expect(result.comparisonData).toEqual(mockStore.comparisonData);
    expect(result.selectedConcern).toEqual(mockStore.selectedConcern);
    expect(result.sessions).toEqual(mockStore.sessions);
  });
});

describe('mapDispatchToProps', () => {
  it('Should call dispatch when createSession is called', () => {
    const mockDispatch = jest.fn();
    const mockParameters = 5;
    const result = mapDispatchToProps(mockDispatch);

    result.createSession(mockParameters);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('Should call dispatch when getSession is called', () => {
    const mockDispatch = jest.fn();
    const mockParameters = 5;
    const result = mapDispatchToProps(mockDispatch);

    result.getSession(mockParameters);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('Should call dispatch when compareSessionData is called', () => {
    const mockDispatch = jest.fn();
    const mockParameters = [5, 3];
    const result = mapDispatchToProps(mockDispatch);

    result.compareSessionData(...mockParameters);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('Should call dispatch when wipeStoreFromSessions is called', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);

    result.wipeStoreFromSessions();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
