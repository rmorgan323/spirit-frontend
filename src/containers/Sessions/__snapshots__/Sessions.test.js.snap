// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sessions
    compareSessionData={[Function]}
    comparisonData={
      Array [
        Object {
          "created": Array [
            Object {
              "date": "2/26/2018",
            },
            Object {
              "date": "2/24/2018",
            },
          ],
        },
        Object {
          "session_id": Array [
            Object {
              "id": 54,
            },
            Object {
              "id": 53,
            },
          ],
        },
        Object {
          "executive": Array [
            Object {
              "x": 1,
              "y": 22,
            },
            Object {
              "x": 0,
              "y": 20,
            },
          ],
        },
        Object {
          "modulation": Array [
            Object {
              "x": 1,
              "y": 18,
            },
            Object {
              "x": 0,
              "y": 12,
            },
          ],
        },
        Object {
          "postural": Array [
            Object {
              "x": 1,
              "y": 15,
            },
            Object {
              "x": 0,
              "y": 8,
            },
          ],
        },
        Object {
          "sensory": Array [
            Object {
              "x": 1,
              "y": 28,
            },
            Object {
              "x": 0,
              "y": 22,
            },
          ],
        },
        Object {
          "social": Array [
            Object {
              "x": 1,
              "y": 18,
            },
            Object {
              "x": 0,
              "y": 13,
            },
          ],
        },
      ]
    }
    createSession={[Function]}
    getSession={[Function]}
    selectedConcern={
      Object {
        "created_at": "2018-02-20T20:08:19.142Z",
        "description": "Concern 1",
        "domain_1": true,
        "domain_2": true,
        "domain_3": false,
        "domain_4": true,
        "domain_5": false,
        "domain_6": false,
        "id": 5,
        "notes": "Some notes.",
        "patient_id": 5,
        "updated_at": "2018-02-20T20:08:19.142Z",
      }
    }
    sessions={
      Array [
        Object {
          "completed": true,
          "concern_id": 5,
          "created_at": "2018-02-20T21:25:45.411Z",
          "id": 10,
          "updated_at": "2018-02-20T21:25:45.411Z",
        },
      ]
    }
    wipeStoreFromSessions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sessions-main-holder"
        >
          <h2>
            Sessions
          </h2>
          <div
            className="session"
          >
            <div
              className="compare-button-holder"
            >
              <label
                className="compare-button-holder"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="checkmark"
                />
              </label>
            </div>
            <p
              className="session-number"
            >
              Session 
              10
            </p>
            <p
              className="session-date"
            >
              <span
                className="date-span"
              >
                Last update: 
              </span>
              February 20, 2018
            </p>
            <p
              className="session-status"
            >
              Completed
            </p>
            <Link
              className="join-link"
              replace={false}
              to="/spirit/sessions/10/view"
            >
              <button
                className="session-result-button"
                onClick={[Function]}
              >
                RESULTS
              </button>
            </Link>
          </div>
          <div
            className="button-holder"
          >
            <button
              className="create-session-button"
              onClick={[Function]}
            >
              Create New Session
            </button>
          </div>
        </div>,
        <div>
          <div
            className="goals-chart"
          >
            <Connect(Component) />
          </div>
          <Connect(Component) />
        </div>,
      ],
      "className": "Sessions",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Sessions
            </h2>,
            Array [
              <div
                className="session"
              >
                <div
                  className="compare-button-holder"
                >
                  <label
                    className="compare-button-holder"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="checkmark"
                    />
                  </label>
                </div>
                <p
                  className="session-number"
                >
                  Session 
                  10
                </p>
                <p
                  className="session-date"
                >
                  <span
                    className="date-span"
                  >
                    Last update: 
                  </span>
                  February 20, 2018
                </p>
                <p
                  className="session-status"
                >
                  Completed
                </p>
                <Link
                  className="join-link"
                  replace={false}
                  to="/spirit/sessions/10/view"
                >
                  <button
                    className="session-result-button"
                    onClick={[Function]}
                  >
                    RESULTS
                  </button>
                </Link>
              </div>,
            ],
            <div
              className="button-holder"
            >
              <button
                className="create-session-button"
                onClick={[Function]}
              >
                Create New Session
              </button>
            </div>,
          ],
          "className": "sessions-main-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sessions",
            },
            "ref": null,
            "rendered": "Sessions",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="compare-button-holder"
                >
                  <label
                    className="compare-button-holder"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="checkmark"
                    />
                  </label>
                </div>,
                <p
                  className="session-number"
                >
                  Session 
                  10
                </p>,
                <p
                  className="session-date"
                >
                  <span
                    className="date-span"
                  >
                    Last update: 
                  </span>
                  February 20, 2018
                </p>,
                <p
                  className="session-status"
                >
                  Completed
                </p>,
                false,
                <Link
                  className="join-link"
                  replace={false}
                  to="/spirit/sessions/10/view"
                >
                  <button
                    className="session-result-button"
                    onClick={[Function]}
                  >
                    RESULTS
                  </button>
                </Link>,
              ],
              "className": "session",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <label
                    className="compare-button-holder"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="checkmark"
                    />
                  </label>,
                  "className": "compare-button-holder",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />,
                      <span
                        className="checkmark"
                      />,
                    ],
                    "className": "compare-button-holder",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "checked": false,
                        "onChange": [Function],
                        "type": "checkbox",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "checkmark",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                  ],
                  "type": "label",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Session ",
                    10,
                  ],
                  "className": "session-number",
                },
                "ref": null,
                "rendered": Array [
                  "Session ",
                  10,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="date-span"
                    >
                      Last update: 
                    </span>,
                    "February 20, 2018",
                  ],
                  "className": "session-date",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Last update: ",
                      "className": "date-span",
                    },
                    "ref": null,
                    "rendered": "Last update: ",
                    "type": "span",
                  },
                  "February 20, 2018",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Completed",
                  "className": "session-status",
                },
                "ref": null,
                "rendered": "Completed",
                "type": "p",
              },
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="session-result-button"
                    onClick={[Function]}
                  >
                    RESULTS
                  </button>,
                  "className": "join-link",
                  "replace": false,
                  "to": "/spirit/sessions/10/view",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "RESULTS",
                    "className": "session-result-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "RESULTS",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="create-session-button"
                  onClick={[Function]}
                >
                  Create New Session
                </button>,
                undefined,
              ],
              "className": "button-holder",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create New Session",
                  "className": "create-session-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Create New Session",
                "type": "button",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="goals-chart"
            >
              <Connect(Component) />
            </div>,
            <Connect(Component) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Component) />,
              "className": "goals-chart",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sessions-main-holder"
          >
            <h2>
              Sessions
            </h2>
            <div
              className="session"
            >
              <div
                className="compare-button-holder"
              >
                <label
                  className="compare-button-holder"
                >
                  <input
                    checked={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="checkmark"
                  />
                </label>
              </div>
              <p
                className="session-number"
              >
                Session 
                10
              </p>
              <p
                className="session-date"
              >
                <span
                  className="date-span"
                >
                  Last update: 
                </span>
                February 20, 2018
              </p>
              <p
                className="session-status"
              >
                Completed
              </p>
              <Link
                className="join-link"
                replace={false}
                to="/spirit/sessions/10/view"
              >
                <button
                  className="session-result-button"
                  onClick={[Function]}
                >
                  RESULTS
                </button>
              </Link>
            </div>
            <div
              className="button-holder"
            >
              <button
                className="create-session-button"
                onClick={[Function]}
              >
                Create New Session
              </button>
            </div>
          </div>,
          <div>
            <div
              className="goals-chart"
            >
              <Connect(Component) />
            </div>
            <Connect(Component) />
          </div>,
        ],
        "className": "Sessions",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Sessions
              </h2>,
              Array [
                <div
                  className="session"
                >
                  <div
                    className="compare-button-holder"
                  >
                    <label
                      className="compare-button-holder"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="checkmark"
                      />
                    </label>
                  </div>
                  <p
                    className="session-number"
                  >
                    Session 
                    10
                  </p>
                  <p
                    className="session-date"
                  >
                    <span
                      className="date-span"
                    >
                      Last update: 
                    </span>
                    February 20, 2018
                  </p>
                  <p
                    className="session-status"
                  >
                    Completed
                  </p>
                  <Link
                    className="join-link"
                    replace={false}
                    to="/spirit/sessions/10/view"
                  >
                    <button
                      className="session-result-button"
                      onClick={[Function]}
                    >
                      RESULTS
                    </button>
                  </Link>
                </div>,
              ],
              <div
                className="button-holder"
              >
                <button
                  className="create-session-button"
                  onClick={[Function]}
                >
                  Create New Session
                </button>
              </div>,
            ],
            "className": "sessions-main-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sessions",
              },
              "ref": null,
              "rendered": "Sessions",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="compare-button-holder"
                  >
                    <label
                      className="compare-button-holder"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="checkmark"
                      />
                    </label>
                  </div>,
                  <p
                    className="session-number"
                  >
                    Session 
                    10
                  </p>,
                  <p
                    className="session-date"
                  >
                    <span
                      className="date-span"
                    >
                      Last update: 
                    </span>
                    February 20, 2018
                  </p>,
                  <p
                    className="session-status"
                  >
                    Completed
                  </p>,
                  false,
                  <Link
                    className="join-link"
                    replace={false}
                    to="/spirit/sessions/10/view"
                  >
                    <button
                      className="session-result-button"
                      onClick={[Function]}
                    >
                      RESULTS
                    </button>
                  </Link>,
                ],
                "className": "session",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <label
                      className="compare-button-holder"
                    >
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="checkmark"
                      />
                    </label>,
                    "className": "compare-button-holder",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          checked={false}
                          onChange={[Function]}
                          type="checkbox"
                        />,
                        <span
                          className="checkmark"
                        />,
                      ],
                      "className": "compare-button-holder",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "checked": false,
                          "onChange": [Function],
                          "type": "checkbox",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "checkmark",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "span",
                      },
                    ],
                    "type": "label",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Session ",
                      10,
                    ],
                    "className": "session-number",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Session ",
                    10,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="date-span"
                      >
                        Last update: 
                      </span>,
                      "February 20, 2018",
                    ],
                    "className": "session-date",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Last update: ",
                        "className": "date-span",
                      },
                      "ref": null,
                      "rendered": "Last update: ",
                      "type": "span",
                    },
                    "February 20, 2018",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Completed",
                    "className": "session-status",
                  },
                  "ref": null,
                  "rendered": "Completed",
                  "type": "p",
                },
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <button
                      className="session-result-button"
                      onClick={[Function]}
                    >
                      RESULTS
                    </button>,
                    "className": "join-link",
                    "replace": false,
                    "to": "/spirit/sessions/10/view",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "RESULTS",
                      "className": "session-result-button",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "RESULTS",
                    "type": "button",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="create-session-button"
                    onClick={[Function]}
                  >
                    Create New Session
                  </button>,
                  undefined,
                ],
                "className": "button-holder",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create New Session",
                    "className": "create-session-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Create New Session",
                  "type": "button",
                },
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="goals-chart"
              >
                <Connect(Component) />
              </div>,
              <Connect(Component) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Component) />,
                "className": "goals-chart",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
